packages/wdio-allure-reporter/src/index.ts

```diff
@@ -1,605 +1,686 @@
+        process.on(events.addArgument, this.addArgument.bind(this))
+    }
+
+    onRunnerStart(runner: RunnerStats) {
+        this._config = runner.config
+        this.capabilities = runner.capabilities
+        this.isMultiremote = runner.isMultiremote || false
+    }
+
+    onSuiteStart(suite: SuiteStats) {
+        if (this._options.useCucumberStepReporter) {
+            if (suite.type === 'feature') {
+                // handle cucumber features as allure "suite"
+                return this.allure.startSuite(suite.title)
+            }
+
+            // handle cucumber scenarii as allure "case" instead of "suite"
+            this.allure.startCase(suite.title)
+            const currentTest = this.allure.getCurrentTest()
+            this.getLabels(suite).forEach(({ name, value }) => {
+                currentTest.addLabel(name, value)
+            })
+            if (suite.description) {
+                this.addDescription(suite)
+            }
+            return this.setCaseParameters(suite.cid)
+        }
+
+        const currentSuite = this.allure.getCurrentSuite()
+        const prefix = currentSuite ? currentSuite.name + ': ' : ''
+        this.allure.startSuite(prefix + suite.title)
+    }
+
+    onSuiteEnd(suite: SuiteStats) {
+        if (this._options.useCucumberStepReporter && suite.type === 'scenario') {
+            // passing hooks are missing the 'state' property
+            suite.hooks = suite.hooks!.map((hook) => {
+                hook.state = hook.state ? hook.state : 'passed'
+                return hook
+            })
+            const suiteChildren = [...suite.tests!, ...suite.hooks]
+            const isPassed = !suiteChildren.some(item => item.state !== 'passed')
+            if (isPassed) {
+                return this.allure.endCase('passed')
+            }
+
+            // A scenario is it skipped if is not passed and every steps/hooks are passed or skipped
+            const isSkipped = suiteChildren.every(item => [PASSED, SKIPPED].indexOf(item.state!) >= 0)
+            if (isSkipped) {
+                return this.allure.endCase(PENDING)
+            }
+
+            // Only close passing and skipped tests because
+            // failing tests are closed in onTestFailed event
+            return
+        }
+
+        this.allure.endSuite()
+    }
+
+    onTestStart(test: TestStats | HookStats) {
+        const testTitle = test.currentTest ? test.currentTest : test.title
+        if (this.isAnyTestRunning() && this.allure.getCurrentTest().name == testTitle) {
+            // Test already in progress, most likely started by a before each hook
+            this.setCaseParameters(test.cid)
+            return
+        }
+
+        if (this._options.useCucumberStepReporter) {
+            return this.allure.startStep(testTitle)
+        }
+
+        this.allure.startCase(testTitle)
+        this.setCaseParameters(test.cid)
+    }
+
+    setCaseParameters(cid?: string) {

```

Comment:
```suggestion
    setCaseParameters(cid: string) {

```

Reply:
cid type in `SuittStats` is `string | undefined`. This is a short hand.